### PR TITLE
Add support for images

### PR DESCRIPTION
**Pull request by OlliZi**

### CODE REVIEW

**Comment by OlliZi** [comment]
Who reviews this:)?

**Comment by OlliZi** [comment]
https://github.com/ChuckerTeam/chucker/issues/64

**Comment by cortinico** [comment]
Hey @OlliZi 
Thanks for the PR :) I'll be happy to do your review. I just ask you to wait a couple of days since I'm on vacation

**Comment by OlliZi** [comment]
What are the next steps (for me)? Can i help you:)?

**Comment by cortinico** [comment]
And sorry for the late reply :(

File: library/src/main/java/com/chuckerteam/chucker/api/internal/ui/transaction/TransactionPayloadFragment.kt [hunk 1]
```diff
@@ -121,6 +130,21 @@ internal class TransactionPayloadFragment : Fragment(), TransactionFragment, Sea
         return true
     }
 
+    private fun udpateImageDataView(imageData: ByteArray?) {
+        val bitmap = if (imageData != null) {
```
Comment (cortinico) [inline library/src/main/java/com/chuckerteam/chucker/api/internal/ui/transaction/TransactionPayloadFragment.kt]:
```suggestion
        val bitmap = imageData?.let {
            BitmapFactory.decodeByteArray(it, 0, it.size)
        }
```

File: library/src/main/res/layout/chucker_fragment_transaction_payload.xml [hunk 2]
```diff
@@ -39,6 +39,14 @@
             android:layout_height="wrap_content"
             android:textIsSelectable="true"/>
 
+        <ImageView
+            android:id="@+id/image"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:contentDescription="image"
+            android:textIsSelectable="true"
```
Comment (cortinico) [inline library/src/main/res/layout/chucker_fragment_transaction_payload.xml]: why this?

File: library/src/main/java/com/chuckerteam/chucker/api/internal/ui/transaction/TransactionPayloadFragment.kt [hunk 3]
```diff
@@ -83,29 +87,34 @@ internal class TransactionPayloadFragment : Fragment(), TransactionFragment, Sea
...
-                    transaction!!.isResponseBodyPlainText
+                    transaction!!.isResponseBodyPlainText,
+                    imageData
                 )
             }
         }
     }
 
-    private fun setText(headersString: String, bodyString: String?, isPlainText: Boolean) {
+    private fun setText(headersString: String, bodyString: String?, isPlainText: Boolean, imageData: ByteArray?) {
```
Comment (cortinico) [inline library/src/main/java/com/chuckerteam/chucker/api/internal/ui/transaction/TransactionPayloadFragment.kt]: Can I ask you to rename this function to `setBody`?

File: library/src/main/java/com/chuckerteam/chucker/api/internal/ui/transaction/TransactionPayloadFragment.kt (hunk 1 above)
Comment (OlliZi) [reply library/src/main/java/com/chuckerteam/chucker/api/internal/ui/transaction/TransactionPayloadFragment.kt]: Done.

File: library/src/main/res/layout/chucker_fragment_transaction_payload.xml (hunk 2 above)
Comment (OlliZi) [reply library/src/main/res/layout/chucker_fragment_transaction_payload.xml]: Done.

File: library/src/main/java/com/chuckerteam/chucker/api/internal/ui/transaction/TransactionPayloadFragment.kt (hunk 3 above)
Comment (OlliZi) [reply library/src/main/java/com/chuckerteam/chucker/api/internal/ui/transaction/TransactionPayloadFragment.kt]: Done.

**Comment by OlliZi** [comment]
Hey:) Thanks for your usefull comments:)

a) You're storing only one ByteArray while ideally you should store 2 (one for the request and one for the response). This is causing an image being shown both in the Request and the Response.
--> Done. The image is only shown in the response tab.

"b) You're always storing transaction.byteData while you should check the content type before storing it. E.g. If the user is receiving video data, you will store it inside the DB and it will not be shown."
--> Done. Add check for an image.

"c) You're still showing a (encoded or binary body omitted) message to the user:"
--> Done. Hide message if an image is present.

**Pull request merged**